### PR TITLE
Week 1 Project Olga Blud

### PR DESCRIPTION
**Pull request by OlgaBlud**
Hello, please take a look at my homework (ecommerce project).
Thank you in advance!

Here is Netlify link :
https://project--storied-nasturtium-34664d.netlify.app/

### CODE REVIEW

File: week1/project/ecommerce/src/App.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  const [category, setCategory] = useState("All products");
+  // console.log(category);
+  return (
+    <div className="App">
+      <h1>PRODUCTS</h1>
+      <CategoryList setCategoryName={setCategory} />
+      <ProductList activeCategory={category} />
+    </div>
+  );
+}
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/App.js:16]: perfect and very clean 🚀

File: week1/project/ecommerce/src/App.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import "./App.css";
+import { useState } from "react";
+import CategoryList from "./components/category-list";
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/App.js]: component Names and file names should be in PascalCase `CategoryList`
✏️need to modify✏️

File: week1/project/ecommerce/src/components/category-list.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  function chooseCategory(name) {
+    const categoryName = name.slice(6);
+    // console.log(categoryName);
+    const checkedCategoryState = (prevState) => {
+      if (categoryName === prevState) {
+        return "All products";
+      } else {
+        return categoryName;
+      }
+    };
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/components/category-list.jsx]: smart 🚀

File: week1/project/ecommerce/src/components/category-list.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      {categoryList.map((category, i) => (
+        <Category
+          name={category}
+          key={i}
+          onClick={() => {
+            chooseCategory(category);
+          }}
+        />
+      ))}
+    </div>
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/components/category-list.jsx]: I like that very clean and readable 👍💥

File: week1/project/ecommerce/src/components/category.jsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+function Category({ name, onClick }) {
+  return (
+    <button className="category-btn" onClick={onClick}>
+      {name}
+    </button>
+  );
+}
+export default Category;
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/components/category.jsx]: Perfect

File: week1/project/ecommerce/src/App.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import ProductList from "./components/product-list";
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/App.js]: same here about naming the file 
✏️need to modify✏️

File: week1/project/ecommerce/src/components/product-list.jsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      : products;
+
+  return (
+    <ul className="products-list">
+      {filteredList.map((item) => {
+        const { id, image, title } = item;
+
+        return <Product key={id} title={title} img={image} />;
+      })}
+    </ul>
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/components/product-list.jsx]: amazing 💥

**Comment by ObadaMoustafa** [comment]
Great work Olga you just need to modify file names as well and that's all about this week 🚀🚀

File: week1/project/ecommerce/src/App.js (hunk 2 above)
Comment (OlgaBlud) [reply week1/project/ecommerce/src/App.js]: Fixed

File: week1/project/ecommerce/src/App.js (hunk 6 above)
Comment (OlgaBlud) [reply week1/project/ecommerce/src/App.js]: fixed

**Comment by OlgaBlud** [comment]
Fixed with file names

**Pull request closed without merging**